stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -128,20 +134,21 @@ class PaymentSheet internal constructor(
     internal interface FlowController {
         fun getPaymentOption(): PaymentOption?
 
-        fun configure(
+        fun configureWithPaymentIntent(
             paymentIntentClientSecret: String,
-            configuration: Configuration,
+            configuration: Configuration? = null,
             callback: ConfigCallback
         )
 
-        fun configure(
-            paymentIntentClientSecret: String,
+        fun configureWithSetupIntent(
+            setupIntentClientSecret: String,
+            configuration: Configuration? = null,
```

Comment:
We'll need variants without the `configuration` parameter

Reply:
The SetupIntent variant will always need a configuration, right? Since we need to attach to a customer?
It looks like in iOS the `Configuration` is not optional anywhere. Do we want to be consistent across platforms? @yuki-stripe

Reply:
You can in fact create & confirm a SetupIntent without a Customer.

I don't have a strong opinion either way - maybe we should make it non-optional as long as it has a default value eg user can pass a bare `Configuration()`?  The majority of integrations will use `configuration`.

Reply:
Added overloading method manually as we don't have an empty constructor for `Configuration` and to keep it consistent with the `PaymentSheet` `@JvmOverloads` methods. Updated documentation.